src/game/AI/ScriptDevAI/scripts/northrend/storm_peaks.cpp

```diff
@@ -223,6 +223,145 @@ bool QuestAccept_npc_injured_miner(Player* pPlayer, Creature* pCreature, const Q
     return false;
 }
 
+/*######
+## npc_ethereal_frostworg
+######*/
+
+enum
+{    // misc
+    SPELL_SUMMON_INFILTRATOR                 = 56325,
+    NPC_STORMFORGED_INFILTRATOR              = 30222,
+
+    FROSTWOLF_MOVE_NORMAL = 1,
+    FROSTWOLF_MOVE_FINAL = 2,
+
+};
+
+enum FrostwolfActions
+{
+    FROSTWOLF_CHANGE_DIRECTION,
+    FROSTWOLF_RESUME_SEARCH,
+    FROSTWOLF_FIND_INFILTRATOR,
+    FROSTWOLF_SUMMON_INFILTRATOR,
+};
+
+static const uint32 ethereal_frostworg_emotes[4][2] =
+{
+    {30909, 30910},
+    {30911, 30912},
+    {30913, 30914},
+    {30915, 30915}
+};
+
+struct npc_ethereal_frostworgAI : public ScriptedAI
+{
+    npc_ethereal_frostworgAI(Creature* creature) : ScriptedAI(creature) {
+        AddCustomAction(FROSTWOLF_FIND_INFILTRATOR, 45s, [&]()
+        {
+            DisableTimer(FROSTWOLF_CHANGE_DIRECTION);
+            MoveToNewPoint(FROSTWOLF_MOVE_FINAL);
+        });
+        AddCustomAction(FROSTWOLF_CHANGE_DIRECTION, 15s, [&]()
+        {
+            m_creature->GetMotionMaster()->Clear();
+            m_creature->GetMotionMaster()->MoveIdle();
+            ResetTimer(FROSTWOLF_RESUME_SEARCH, 3s);
+            if (m_emoteCounter < 3)
+                ResetTimer(FROSTWOLF_CHANGE_DIRECTION, 15s);
+        });
+        AddCustomAction(FROSTWOLF_RESUME_SEARCH, 0s, [&]()
+        {
+            switch (m_emoteCounter)
+            {
+                case 0:
+                case 1:
+                case 2:
+                {
+                    if (!m_creature->IsInCombat())
+                        DoBroadcastText(ethereal_frostworg_emotes[m_emoteCounter][urand(0,1)], m_creature);
+                    break;
+                }
+                case 3: break;
+                default: m_creature->ForcedDespawn(); break;
+            }
+            ++m_emoteCounter;
+            
+            MoveToNewPoint(FROSTWOLF_MOVE_NORMAL);
+        });
+    }
+
+    uint32 m_emoteCounter = 0;
+    bool m_summoned = false;
+
+    void MoveToNewPoint(uint32 type)
+    {
+        float dist = type == 1 ? 50.f : 20.f;
+        float x, y, z = 0.f;
+        while (abs(z - m_creature->GetPositionZ()) > 5)
+            m_creature->GetRandomPoint(m_creature->GetPositionX(), m_creature->GetPositionY(), m_creature->GetPositionZ(), dist, x, y, z, 10.f);
```

Comment:
Not sure about the minimum point distance, I guesstimated it to 10 because it feels to fit to the reference